components/org.wso2.carbon.identity.application.authenticator.kakao/src/main/java/org/wso2/carbon/identity/application/authenticator/kakao/KakaoCustomAuthenticator.java

```diff
@@ -76,21 +123,251 @@ public List<Property> getConfigurationProperties() {
         clientSecret.setConfidential(true);
         clientSecret.setDescription("Enter client secret value");
         configProperties.add(clientSecret);
 
         Property callbackUrl = new Property();
-        callbackUrl.setName(KakaoCustomAuthenticatorConstants.CALLBACK_URL);
+        callbackUrl.setName(CALLBACK_URL);
         callbackUrl.setDisplayName("Callback Url");
         callbackUrl.setRequired(true);
         callbackUrl.setDescription("Enter callback url");
         configProperties.add(callbackUrl);
 
         return configProperties;
     }
+
+    /**
+     * Check whether the authentication is based on API.
+     *
+     * @return true since API based authentication is supported.
+     */
+    @Override
+    public boolean isAPIBasedAuthenticationSupported() {
+
+        return true;
+    }
+
+    /**
+     * Get the i18n key defined to represent the authenticator name.
+     *
+     * @return the 118n key.
+     */
+    @Override
+    public String getI18nKey() {
+
+        return AUTHENTICATOR_I18N_KEY;
+    }
+
+
+    /**
+     * This method is responsible for obtaining authenticator-specific data needed to
+     * initialize the authentication process within the provided authentication context.
+     *
+     * @param context The authentication context containing information about the current authentication attempt.
+     * @return An {@code Optional} containing an {@code AuthenticatorData} object representing the initiation data.
+     * If the initiation data is available, it is encapsulated within the {@code Optional}; otherwise,
+     * an empty {@code Optional} is returned.
+     */
+    @Override
+    public Optional<AuthenticatorData> getAuthInitiationData(AuthenticationContext context) {
+
+        AuthenticatorData authenticatorData = new AuthenticatorData();
+        authenticatorData.setName(getName());
+        authenticatorData.setDisplayName(getFriendlyName());
+        authenticatorData.setI18nKey(getI18nKey());
+        if (context.getExternalIdP() != null) {
+            authenticatorData.setIdp(context.getExternalIdP().getIdPName());
+        }
+
+        List<String> requiredParameterList = new ArrayList<>();
+        if (isTrustedTokenIssuer(context)) {
+            requiredParameterList.add(ACCESS_TOKEN_PARAM);
+            requiredParameterList.add(ID_TOKEN_PARAM);
+            authenticatorData.setPromptType(FrameworkConstants.AuthenticatorPromptType.INTERNAL_PROMPT);
+            authenticatorData.setAdditionalData(getAdditionalData(context, true));
+        } else {
+            requiredParameterList.add(OAUTH2_GRANT_TYPE_CODE);
+            requiredParameterList.add(OAUTH2_PARAM_STATE);
+            authenticatorData.setPromptType(FrameworkConstants.AuthenticatorPromptType.REDIRECTION_PROMPT);
+            authenticatorData.setAdditionalData(getAdditionalData(context, false));
+        }
+        authenticatorData.setRequiredParams(requiredParameterList);
+
```

Comment:
Don't we have a requirement to set any authenticator messages here? Please refer [here](https://github.com/wso2-extensions/identity-outbound-auth-oidc/blob/a09f38d618d1c56ec0eb7ba442ab2d77a71dcc6b/components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java#L1830-L1832).

Reply:
We are only calling the `isTrustedTokenIssuer` and `getAdditionalData` functions of this class inside this function. These functions don't set any authenticator messages therefore at this point there shouldn't be any authenticator messages generated during this function in the context.

Reply:
Added this since on inspection it was noticed that we should add the authenticator message here as it's currently in the context only.